tests/emit.spec.ts

```diff
@@ -78,23 +78,73 @@ describe('emitted', () => {
     expect(wrapper.emitted().hello[1]).toEqual(['foo', 'bar'])
   })
 
-  it('should not propagate child events', () => {
+  it('should propagate child native events', async () => {
     const Child = defineComponent({
-      name: 'Child',
-      setup(props, { emit }) {
+      name: 'Button',
+      setup(props) {
+        return () => h('div', [h('input')])
+      }
+    })
+
+    const Parent = defineComponent({
+      name: 'Parent',
+      setup() {
         return () =>
           h('div', [
-            h('button', { onClick: () => emit('hello', 'foo', 'bar') })
+            h(Child, {
+              onClick: (event: Event) => event.stopPropagation()
+            })
           ])
       }
     })
 
+    const GrandParent: FunctionalComponent<{ level: number }> = (
+      props,
+      ctx
+    ) => {
+      return h(`h${props.level}`, [h(Parent)])
+    }
+
+    const wrapper = mount(GrandParent)
+    const parentWrapper = wrapper.findComponent(Parent)
+    const childWrapper = wrapper.findComponent(Child)
+    const input = wrapper.find('input')
+
+    expect(wrapper.emitted()).toEqual({})
+    expect(parentWrapper.emitted()).toEqual({})
+    expect(childWrapper.emitted()).toEqual({})
+
+    await input.trigger('click')
+
+    // Propagation should stop at Parent
+    expect(childWrapper.emitted().click).toHaveLength(1)
+    expect(parentWrapper.emitted().click).toEqual(undefined)
+    expect(wrapper.emitted().click).toEqual(undefined)
+
+    input.setValue('hey')
+
+    expect(childWrapper.emitted().input).toHaveLength(1)
+    expect(parentWrapper.emitted().input).toHaveLength(1)
+    expect(wrapper.emitted().input).toHaveLength(1)
+  })
+
+  it('should not propagate child custom events', () => {
+    const Child = defineComponent({
+      name: 'Child',
+      emits: ['hi'],
+      setup(props, { emit }) {
+        return () =>
+          h('div', [h('button', { onClick: () => emit('hi', 'foo', 'bar') })])
+      }
+    })
+
     const Parent = defineComponent({
       name: 'Parent',
+      emits: ['hello'],
```

Comment:
Please add this test

It's the one the OP of the original issue had problems with

```ts
  it('captures composition event', async () => {
    const useCommonBindings = () => {
      const onCompositionStart = (evt: CompositionEvent) => {
        const instance = getCurrentInstance()!
        instance.emit('compositionStart', evt)
      }
      return { onCompositionStart }
    }
    const IxInput = defineComponent({
      setup() {
        return useCommonBindings()
      },
      template: `<input @compositionstart="(evt) => $emit('compositionStart', evt)" />`
    })

    const wrapper = mount({
      components: { IxInput },
      template: `<ix-input />`
    })

    await wrapper.find('input').trigger('compositionstart')

    expect(wrapper.emitted().compositionstart).not.toBe(undefined)
  })
```

Reply:
Added in 1ba4e59